File: leetcode/two_sum/twoSum.cpp
#include <map>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        map<int, int> already_looked;
        already_looked[nums[0]] = 0;
        for (int i = 1; i < nums.size(); i++)
        {

            if (already_looked.find(target - nums[i]) == already_looked.end())
            {
                already_looked[nums[i]] = i;
            }
            else
            {
                return {already_looked[target - nums[i]], i};
            }
        }
        return {};
    }
};

File: leetcode/two_sum/two_sum.py
from types import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        already_looked = dict()
        for i, num in enumerate(nums):
            if (target - num) in already_looked:
                return [already_looked[target - num], i]
            else:
                already_looked[num] = i
